Directory.Build.props

```diff
@@ -0,0 +1,28 @@
+<Project>
+
+  <PropertyGroup>
+    <!--
+    NOTE: This .props file assumes it is only imported by projects that use the SDK="Microsoft.NET.Sdk" in their project.
+    We are currently on too old of an SDK to detect - it doesn't have https://github.com/dotnet/sdk/pull/1242.
+    But once we move to a new version, we can remove this property setting, and everything else will work correctly.
+    -->
+    <UsingMicrosoftNETSdk>true</UsingMicrosoftNETSdk>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(UsingMicrosoftNETSdk)' == 'true'">
+    <UsingNETSdkCompiler>true</UsingNETSdkCompiler>
+    <DisableBuildToolsRoslynVersion>true</DisableBuildToolsRoslynVersion>
+  </PropertyGroup>
+
+  <Import Project="dir.props" />
+
+  <PropertyGroup Condition="'$(UsingMicrosoftNETSdk)' == 'true'">
+    <!--
+    The Microsoft.NET.Sdk changed the definition of what it assumes $(BaseIntermediateOutputPath) is.
+    In the "old" .csproj files, BaseIntermediateOutputPath could be shared with many projects.
+    In the SDK-based .csproj files, BaseIntermediateOutputPath is assumed to be project-specific.
+    -->
+    <BaseIntermediateOutputPath>$(IntermediateOutputPath)</BaseIntermediateOutputPath>
```

Comment:
Can we not just set this in dir.props?

Reply:
Not really.  That would cause ripple changes to all the places that currently use `$(BaseIntermediateOutputPath)`.  Looking, there appears to be a few places in BuildTools that use it.  For example, `BuildVersion.targets` uses it for the `$(BuildVersionFilePath)` property.

For non-SDK projects (like our build.proj and dir.proj files), `$(BaseIntermediateOutputPath)` == `$(RepoRoot)\Bin\obj`.
This causes a problem for SDK-based projects (or really any MSBuild v15 projects), since the definition of `$(BaseIntermediateOutputPath)` changed to allow for package managers to lay down `.props/.targets` files that are automatically imported.  See the comments [here](https://github.com/Microsoft/msbuild/blob/38fe952a2ec5885b922e8581d33aa09e0904cf8b/src/Tasks/Microsoft.Common.props#L45-L61).

For SDK-based projects, `$(BaseIntermediateOutputPath)` MUST be unique per project and if you want to change it from `$ProjectDirectory)\obj`, you MUST set it before Microsoft.Common.props is imported.

Reply:
Sigh. Ok.